### PR TITLE
[9.0][FIX] travis: split up travis jobs

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Same patch as https://github.com/OCA/OCB/pull/722.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by pedrobaeza** [comment]
I don't see the exclusion of l10n_generic_coa.

File: .travis.yml [hunk 1]
```diff
@@ -19,7 +19,12 @@ python:
   - "2.7"
 
 env:
-  - VERSION="9.0" ODOO_REPO="OCA/OCB" LINT_CHECK="0" TESTS="1" WEBSITE_REPO="1"
+  global:
+  - VERSION="9.0" ODOO_REPO="OCA/OCB" LINT_CHECK="0" WEBSITE_REPO="1"
+  matrix:
+  - export INCLUDE=applications
```
Comment (pedrobaeza) [inline .travis.yml]: Remove the word `export` as done in 8.0

**Comment by MiquelRForgeFlow** [comment]
I don't know what happens with the "no_aplications" job 😥

@pedrobaeza I decided to not exclude the l10n_generic_coa when I saw that the problem is in the account demo data. Every l10n_ module that creates a coa will fail and excluding them is not the solution.

**Comment by StefanRijnhart** [comment]
What if we consider l10n_generic_coa an application instead of a localization module, as it needs to be installed for account's tests?

**Comment by MiquelRForgeFlow** [comment]
Reproduced the problem in local. The problem is that l10n_generic_coa is executed as --init two times, and this is incorrect. One way to solve this could be to consider the l10n_generic_coa as an application as you said (but that would mean to rewrite the MQT with several exceptions, and I think it will look ugly). Another approach for which I am more inclined is to rewrite the MQT but just ensure one module cannot have two times an --init installation, and maybe for the second time use --update option.

**Comment by StefanRijnhart** [comment]
I think an exception for l10n_generic_coa is acceptible as there are not a lot of modules that do not survive multiple initializations but kudos if you have a more elegant solution.

**Comment by MiquelRForgeFlow** [comment]
Well, the elegant solution depends of knowing how to obtain the full list of the preinstalled modules after they are preinstalled, which I don't know how to do. Probable it means to run a command in the subprocess or access to the database in some way. Maybe @hbrunn could provide a hint.

If not, then I will go with the exception way.

**Comment by StefanRijnhart** [comment]
And that does not sound so elegant in the first place ;-)

**Comment by MiquelRForgeFlow** [comment]
One solution: https://github.com/OCA/maintainer-quality-tools/pull/541

The other solution: https://github.com/OCA/maintainer-quality-tools/pull/543

@StefanRijnhart @pedrobaeza

**Comment by moylop260** [comment]
For `INCLUDE="localization"` you will have errors using `l10n_generic_coa` this module is installed from [account](https://github.com/odoo/odoo/blob/dea07fc29116c36c8a3773eec0fce526a972ba08/addons/account/__init__.py#L32) using `company.country` field from that the database is created thanks to that this field was added to main form to create new database:
 - country

Then if you use country = México, you will have `l10n_mx` pre-installed without `generic_coa`

The unique way that I found to bypass this behaivour is assigning the company.country value in a module without depends (to have major priority than `account` module):
 - https://github.com/Vauxoo/server-tools/tree/11.0/company_country

Then for my CI I use this project as dependency to `oca_dependencies`, adds this module as dependency of main module app and assign a environment variable `COUNTRY="mx"` from `.travis.yml` file and this works for tests CI and production, because if the module is declared and the variable is not assigned then a error is raised then the SYSADMIN report it and we request adds the correct environment variable)

I don't know if there is other way to bypass it but is a recommendation of Quentin from:
 - https://github.com/OCA/maintainer-quality-tools/issues/412#issuecomment-272906192
 - https://github.com/odoo/odoo/pull/15004
 - https://github.com/OCA/maintainer-quality-tools/issues/412

Then we could use UNITEST="1" in order to install a localization for each country and assign `company.country` value early (before account module) in order to test it correctly.

**Comment by StefanRijnhart** [comment]
@moylop260 I think we should settle for l10n_generic_coa for OCB

**Comment by moylop260** [comment]
Did you mean is delete the invocation for that hook using a commit for OCA/OCB project?

What about use a sed from `.travis.yml` like as:
 - `find ${HOME} -name __openerp__.py -exec sed -i  "/'_auto_install_l10n'/d" {} \;` 
and this hook never will be executed.

screen shot 2018-04-19 at 13 09 43


I don't know.

**Review by MiquelRForgeFlow — COMMENTED** [review]
:+1:

**Comment by moylop260** [comment]
First build is testing the following modules:
`app_modules = set(["account", "account_accountant", "account_bank_statement_import", "account_full_reconcile", "analytic", "auth_crypt", "auth_signup", "barcodes", "base", "base_action_rule", "base_import", "base_setup", "board", "bus", "calendar", "crm", "decimal_precision", "document", "event", "event_sale", "fetchmail", "fleet", "gamification", "gamification_sale_crm", "hr", "hr_attendance", "hr_contract", "hr_equipment", "hr_expense", "hr_gamification", "hr_holidays", "hr_recruitment", "hr_timesheet", "hr_timesheet_sheet", "im_livechat", "link_tracker", "lunch", "mail", "mail_tip", "marketing", "mass_mailing", "mrp", "mrp_repair", "note", "payment", "payment_transfer", "point_of_sale", "portal", "portal_gamification", "portal_sale", "portal_stock", "procurement", "procurement_jit", "product", "project", "project_issue", "project_timesheet", "purchase", "rating", "report", "resource", "sale", "sale_crm", "sale_expense", "sale_mrp", "sale_service", "sale_stock", "sale_timesheet", "sales_team", "stock", "stock_account", "survey", "survey_crm", "utm", "web", "web_calendar", "web_diagram", "web_editor", "web_kanban", "web_kanban_gauge", "web_planner", "web_settings_dashboard", "web_tip", "web_view_editor", "website", "website_blog", "website_event", "website_event_sale", "website_forum", "website_links", "website_livechat", "website_mail", "website_mass_mailing", "website_partner", "website_payment", "website_portal", "website_portal_sale", "website_project_issue", "website_sale", "website_sale_stock", "website_slides", "website_theme_install"])`

Second one:
`noapp_modules = ["account_analytic_default", "account_asset", "account_budget", "account_cancel", "account_cash_basis_base_account", "account_check_printing", "account_extra_reports", "account_tax_adjustments", "account_tax_cash_basis", "account_tax_exigible", "account_tax_python", "account_test", "account_voucher", "anonymization", "association", "auth_ldap", "auth_oauth", "base_gengo", "base_geolocalize", "base_iban", "base_import_module", "base_vat", "claim_from_delivery", "crm_claim", "crm_partner_assign", "crm_project_issue", "delivery", "google_account", "google_calendar", "google_drive", "google_spreadsheet", "hr_payroll", "hr_payroll_account", "hw_blackbox_be", "hw_proxy", "hw_scale", "im_odoo_support", "marketing_campaign", "marketing_campaign_crm_demo", "membership", "mrp_byproduct", "mrp_operations", "note_pad", "pad", "pad_project", "payment_adyen", "payment_authorize", "payment_buckaroo", "payment_ogone", "payment_paypal", "payment_sips", "pos_cache", "pos_discount", "pos_mercury", "pos_reprint", "pos_restaurant", "product_email_template", "product_expiry", "product_extended", "product_margin", "product_uos", "product_visible_discount", "project_issue_sheet", "purchase_requisition", "rating_project", "rating_project_issue", "report_intrastat", "report_webkit", "sale_layout", "sale_margin", "sale_order_dates", "stock_calendar", "stock_dropshipping", "stock_landed_costs", "stock_picking_wave", "subscription", "theme_default", "warning", "web_analytics", "website_crm", "website_crm_claim", "website_crm_partner_assign", "website_customer", "website_event_questions", "website_event_track", "website_form", "website_forum_doc", "website_gengo", "website_google_map", "website_hr", "website_hr_recruitment", "website_issue", "website_mail_channel", "website_membership", "website_project_issue_sheet", "website_quote", "website_rating_project_issue", "website_sale_delivery", "website_sale_digital", "website_sale_options", "website_twitter"]`

Third one:
`localization_modules = set(["l10n_ae", "l10n_ar", "l10n_at", "l10n_au", "l10n_be", "l10n_be_hr_payroll", "l10n_be_hr_payroll_account", "l10n_be_intrastat", "l10n_be_invoice_bba", "l10n_bo", "l10n_br", "l10n_ca", "l10n_ch", "l10n_cl", "l10n_cn", "l10n_cn_small_business", "l10n_cn_standard", "l10n_co", "l10n_cr", "l10n_de", "l10n_de_skr03", "l10n_de_skr04", "l10n_do", "l10n_ec", "l10n_es", "l10n_et", "l10n_eu_service", "l10n_fr", "l10n_fr_certification", "l10n_fr_fec", "l10n_fr_hr_payroll", "l10n_fr_pos_cert", "l10n_fr_sale_closing", "l10n_generic_coa", "l10n_gr", "l10n_gt", "l10n_hn", "l10n_hr", "l10n_hu", "l10n_in", "l10n_in_hr_payroll", "l10n_in_schedule6", "l10n_it", "l10n_jp", "l10n_lu", "l10n_ma", "l10n_multilang", "l10n_mx", "l10n_nl", "l10n_no", "l10n_nz", "l10n_pa", "l10n_pe", "l10n_pl", "l10n_pt", "l10n_ro", "l10n_sa", "l10n_sg", "l10n_si", "l10n_syscohada", "l10n_th", "l10n_tr", "l10n_uk", "l10n_us", "l10n_uy", "l10n_ve", "l10n_vn"])`

All odoo modules are:
`all = set(["account", "account_accountant", "account_analytic_default", "account_asset", "account_bank_statement_import", "account_budget", "account_cancel", "account_cash_basis_base_account", "account_check_printing", "account_extra_reports", "account_full_reconcile", "account_tax_adjustments", "account_tax_cash_basis", "account_tax_exigible", "account_tax_python", "account_test", "account_voucher", "analytic", "anonymization", "association", "auth_crypt", "auth_ldap", "auth_oauth", "auth_signup", "barcodes", "base_action_rule", "base_gengo", "base_geolocalize", "base_iban", "base_import", "base_import_module", "base_setup", "base_vat", "board", "bus", "calendar", "claim_from_delivery", "crm", "crm_claim", "crm_partner_assign", "crm_project_issue", "decimal_precision", "delivery", "document", "event", "event_sale", "fetchmail", "fleet", "gamification", "gamification_sale_crm", "google_account", "google_calendar", "google_drive", "google_spreadsheet", "hr", "hr_attendance", "hr_contract", "hr_equipment", "hr_expense", "hr_gamification", "hr_holidays", "hr_payroll", "hr_payroll_account", "hr_recruitment", "hr_timesheet", "hr_timesheet_sheet", "hw_blackbox_be", "hw_escpos", "hw_posbox_homepage", "hw_posbox_upgrade", "hw_proxy", "hw_scale", "hw_scanner", "hw_screen", "im_livechat", "im_odoo_support", "l10n_ae", "l10n_ar", "l10n_at", "l10n_au", "l10n_be", "l10n_be_hr_payroll", "l10n_be_hr_payroll_account", "l10n_be_intrastat", "l10n_be_invoice_bba", "l10n_bo", "l10n_br", "l10n_ca", "l10n_ch", "l10n_cl", "l10n_cn", "l10n_cn_small_business", "l10n_cn_standard", "l10n_co", "l10n_cr", "l10n_de", "l10n_de_skr03", "l10n_de_skr04", "l10n_do", "l10n_ec", "l10n_es", "l10n_et", "l10n_eu_service", "l10n_fr", "l10n_fr_certification", "l10n_fr_fec", "l10n_fr_hr_payroll", "l10n_fr_pos_cert", "l10n_fr_sale_closing", "l10n_generic_coa", "l10n_gr", "l10n_gt", "l10n_hn", "l10n_hr", "l10n_hu", "l10n_in", "l10n_in_hr_payroll", "l10n_in_schedule6", "l10n_it", "l10n_jp", "l10n_lu", "l10n_ma", "l10n_multilang", "l10n_mx", "l10n_nl", "l10n_no", "l10n_nz", "l10n_pa", "l10n_pe", "l10n_pl", "l10n_pt", "l10n_ro", "l10n_sa", "l10n_sg", "l10n_si", "l10n_syscohada", "l10n_th", "l10n_tr", "l10n_uk", "l10n_us", "l10n_uy", "l10n_ve", "l10n_vn", "link_tracker", "lunch", "mail", "mail_tip", "marketing", "marketing_campaign", "marketing_campaign_crm_demo", "mass_mailing", "membership", "mrp", "mrp_byproduct", "mrp_operations", "mrp_repair", "note", "note_pad", "pad", "pad_project", "payment", "payment_adyen", "payment_authorize", "payment_buckaroo", "payment_ogone", "payment_paypal", "payment_sips", "payment_transfer", "point_of_sale", "portal", "portal_gamification", "portal_sale", "portal_stock", "pos_cache", "pos_discount", "pos_mercury", "pos_reprint", "pos_restaurant", "procurement", "procurement_jit", "product", "product_email_template", "product_expiry", "product_extended", "product_margin", "product_uos", "product_visible_discount", "project", "project_issue", "project_issue_sheet", "project_timesheet", "purchase", "purchase_requisition", "rating", "rating_project", "rating_project_issue", "report", "report_intrastat", "report_webkit", "resource", "sale", "sale_crm", "sale_expense", "sale_layout", "sale_margin", "sale_mrp", "sale_order_dates", "sale_service", "sale_stock", "sale_timesheet", "sales_team", "stock", "stock_account", "stock_calendar", "stock_dropshipping", "stock_landed_costs", "stock_picking_wave", "subscription", "survey", "survey_crm", "theme_bootswatch", "theme_default", "utm", "warning", "web", "web_analytics", "web_calendar", "web_diagram", "web_editor", "web_kanban", "web_kanban_gauge", "web_planner", "web_settings_dashboard", "web_tip", "web_view_editor", "website", "website_blog", "website_crm", "website_crm_claim", "website_crm_partner_assign", "website_customer", "website_event", "website_event_questions", "website_event_sale", "website_event_track", "website_form", "website_forum", "website_forum_doc", "website_gengo", "website_google_map", "website_hr", "website_hr_recruitment", "website_issue", "website_links", "website_livechat", "website_mail", "website_mail_channel", "website_mass_mailing", "website_membership", "website_partner", "website_payment", "website_portal", "website_portal_sale", "website_project_issue", "website_project_issue_sheet", "website_quote", "website_rating_project_issue", "website_sale", "website_sale_delivery", "website_sale_digital", "website_sale_options", "website_sale_stock", "website_slides", "website_theme_install", "website_twitter"])`

FYI `all - (localization_modules | noapp_modules | app_modules)`
Result: `['hw_escpos', 'hw_posbox_homepage', 'hw_posbox_upgrade', 'hw_scanner', 'hw_screen', 'theme_bootswatch']`

The modules `hw_*` and `theme_bootswatch` are explicitly excluded from `.travis.yml`:
https://github.com/OCA/OCB/blob/adc53b59eedc78766a057cc9d40bcfba4866e0fe/.travis.yml#L37

Then this is working fine for me.

**Comment by StefanRijnhart** [comment]
@moylop260 Thanks for your work, but I'm a bit sad you merged your own work so quickly without waiting for my input or approval. I have been actively involved with all of these PRs. For instance, why did you disable the test crawl?

**Comment by MiquelRForgeFlow** [comment]
I also wanted to give my input on the topic :/

**Comment by moylop260** [comment]
Hello @StefanRijnhart 
I merged the work of @mreficent (This is not my PR authored)
I just added little changes to make it greenfy and previously I was authorized to make it from the approval of the author.

Then I merged it

But I don't have problem in revert it if you think that needs it.
Or we can work your feedback from other new PR without problem.

What do you think?

**Pull request merged**